router/functionHandler.go

```diff
@@ -183,9 +187,8 @@ func (roundTripper RetryingRoundTripper) RoundTrip(req *http.Request) (resp *htt
 		// just retry after backing off for timeout period.
 		if serviceUrlFromExecutor {
 			log.Printf("request to %s errored out. backing off for %v before retrying",
-				req.URL.Host, timeout)
-			timeout *= time.Duration(2)
-			time.Sleep(timeout)
+				req.URL.Host, roundTripper.funcHandler.tsRoundTripperParams.timeout)
+			time.Sleep(roundTripper.funcHandler.tsRoundTripperParams.timeout * time.Duration(2))
```

Comment:
This is actually not the same as timeout*=2; you now have a constant sleep through all iterations of the loop.  What you want is something like:

```
<before the loop>
timeout := roundTripper.funcHandler.roundTripperParams.timeout

<in the loop>
...
   timeout *= time.Duration(2)
   time.Sleep(timeout)
```

Reply:
BTW if you make the "2" configurable as ROUNDTRIP_TIMEOUT_EXPONENT or something, that timeout-doubling behaviour also becomes configurable, so you could set it to 1 for much shorter timeouts (i.e. quicker failures, in case that's what you want).

Reply:
Done.